azerothcore/azerothcore-wotlk
# Fix incorrect pathing of caster mobs when their casts fail due to LOS

## Changes Proposed:
Changes the way caster mobs chase you after you move out of their line of sight. Originally when one would hide behind something while still not being too far away, the caster mob would usually move left or right, seemingly unpredictably, to find a spot that is in range and with a clear line of sight. They would only ever try to close in on you if you were completely out of their range.

The change here is that it makes the caster mobs run straight at you after hiding behind something until a clear line of sight is reached. Which is more in line with blizzlike behavior.

## Issues Addressed:
- #3740
- #7268
- #5410

## SOURCE:

## Tests Performed:

- Tested in places referred to by the issues above.
- Moonbrook mages
- Bloodmyst blood elves
- Inside Karazhan


## How to Test the Changes:


1. Pull a caster mob anywhere in the game and run behind something that would block the line of sight. They should run straight towards you until they reach a clear line of sight.

## Known Issues and TODO List:




## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
